docs/source/plug-in.md

```diff
@@ -0,0 +1,26 @@
+## Other modes
+
+The `KernelGatewayApp.api` can be set to the name of any public module supplying a personality. This allows for suporting alternate kernel communications mechanisms.
+
+### Factory function
+`create_personality(*args, **kwargs)`
+
+The only function required in the module, it is passed a named argument `parent` whose value is the kernel gateway application. The function must return an object implementing the following lifecycle and delegated functions:
+
+### Lifecycle functions
```

Comment:
A couple things:

- The doc should recommend that the class derive from LoggingConfigurable.
- It's not clear that the lifecycle nor delegated functions are actually methods on the instance returned. 

I think splitting up the sections will make it hard for readers to envision what exactly they need to implement. Maybe instead you could show a complete, empty module template with docstrings on the functions describing what they need to do, like:

```
from traitlets.config.configurable import LoggingConfigurable

class TemplatePersonality(LoggingConfigurable):
    def init_configurables(self):
        """Put docstring here."""
        pass 

    def shutdown(self):
        """Put docstring here."""
        pass 

    def create_request_handlers(self):
        """Put docstring here."""
        pass 

    def should_seed_cell(self, code):
        """Put docstring here."""
        pass

def create_personality(self, parent):
    return TemplatePersonality(parent=parent)
```